src/compiler/transformers/classFields.ts

```diff
@@ -1074,22 +911,20 @@ namespace ts {
             if (shouldTransformPrivateElements && isPrivateIdentifier(propertyName)) {
                 const privateIdentifierInfo = accessPrivateIdentifier(propertyName);
                 if (privateIdentifierInfo) {
-                    switch (privateIdentifierInfo.placement) {
-                        case PrivateIdentifierPlacement.InstanceField: {
+                    if(!privateIdentifierInfo.kind) {
+                        if (!privateIdentifierInfo.isStatic) {
                             return createPrivateInstanceFieldInitializer(
                                 receiver,
                                 visitNode(property.initializer, visitor, isExpression),
-                                privateIdentifierInfo.weakMapName
+                                privateIdentifierInfo.state
                             );
                         }
-                        case PrivateIdentifierPlacement.StaticField: {
+                        else {
                             return createPrivateStaticFieldInitializer(
-                                privateIdentifierInfo.variableName,
+                                privateIdentifierInfo.writeFunction,

```

Comment:
I understand having a unified name helps with the creation of helper calls but I find it a bit confusing in places like this. I'm not sure if our internal data structures should be one-to-one to the emit helpers, especially considering helpers are polymorphic.

Reply:
I definitely agree that there is room for simplification in the old code. But I'd prefer pushing some complexity into `createPrivateIdentifierAssignment` and `createPrivateIdentifierAccessHelper` and have `PrivateIdentifierInfo` types to have more clear properties based on `kind`.